### PR TITLE
Upgrade test dependencies & fix Falcon warning

### PR DESCRIPTION
**Pull request by seratch**
This pull request upgrades the dependencies in testing. Also, it resolves a deprecation warning in Falcon 3.x.

### Category (place an `x` in each of the `[ ]`)

* [ ] `slack_bolt.App` and/or its core components
* [ ] `slack_bolt.async_app.AsyncApp` and/or its core components
* [x] Adapters in `slack_bolt.adapter`
* [ ] Document pages under `/docs`
* [x] Others

## Requirements (place an `x` in each `[ ]`)

Please read the [Contributing guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and [Code of Conduct](https://slackhq.github.io/code-of-conduct) before creating this issue or pull request. By submitting, you are agreeing to those rules.

* [x] I've read and understood the [Contributing Guidelines](https://github.com/slackapi/bolt-python/blob/main/.github/contributing.md) and have done my best effort to follow them.
* [x] I've read and agree to the [Code of Conduct](https://slackhq.github.io/code-of-conduct).
* [x] I've run `./scripts/install_all_and_run_tests.sh` after making the changes.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -63,36 +63,40 @@
...
-            "websocket_client>=1,<1.2.2",
+            # Note: 1.2.2 has a regression (https://github.com/websocket-client/websocket-client/issues/769)
+            "websocket_client>=1.2.3,<2",
         ],
         # pip install -e ".[testing_without_asyncio]"
         "testing_without_asyncio": test_dependencies,
         # pip install -e ".[testing]"
         "testing": async_test_dependencies,
+        # pip install -e ".[adapter_testing]"
+        "adapter_testing": [
```
Comment (seratch) [inline setup.py:91]: Organized the optional dependencies by moving test-related ones to this group

File: slack_bolt/adapter/falcon/resource.py [hunk 2]
```diff
@@ -50,7 +50,11 @@ def _to_bolt_request(self, req: Request) -> BoltRequest:
         )
 
     def _write_response(self, bolt_resp: BoltResponse, resp: Response):
-        resp.body = bolt_resp.body
+        if falcon_version.__version__.startswith("2."):
+            resp.body = bolt_resp.body
```
Comment (seratch) [inline slack_bolt/adapter/falcon/resource.py:54]: `response.text` does not exist in 2.x

File: slack_bolt/adapter/falcon/resource.py [hunk 3]
```diff
@@ -50,7 +50,11 @@ def _to_bolt_request(self, req: Request) -> BoltRequest:
...
+        else:
+            resp.text = bolt_resp.body
```
Comment (seratch) [inline slack_bolt/adapter/falcon/resource.py:56]: `response.body` is deprecated in 3.x

File: tests/adapter_tests/falcon/test_falcon.py [hunk 4]
```diff
@@ -18,6 +20,13 @@
 from tests.utils import remove_os_env_temporarily, restore_os_env
 
 
+def new_falcon_app():
+    if falcon.version.__version__.startswith("2."):
+        return falcon.API()
```
Comment (seratch) [inline tests/adapter_tests/falcon/test_falcon.py:25]: `falcon.API` is deprecated in 3.x. However, the new recommended way `falcon.App` does not exist in 2.x

**Review by seratch — COMMENTED** [review]
comments for reviewers

**Comment by seratch** [comment]
👋 Let me know if you have any comments on this

**Review by srajiang — APPROVED** [review]
Looks good!

**Comment by seratch** [comment]
Thanks for the review!

**Pull request merged**